spring/integration/pubsub/src/main/java/com/example/spring/pubsub/PubsubApplication.java

```diff
@@ -0,0 +1,153 @@
+package com.example.spring.pubsub;
+
+import com.google.pubsub.v1.Subscription;
+import com.google.pubsub.v1.SubscriptionName;
+import com.google.pubsub.v1.Topic;
+import com.google.pubsub.v1.TopicName;
+import java.io.IOException;
+
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.protobuf.ByteString;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.gcp.pubsub.PubsubAdmin;
+import org.springframework.cloud.gcp.pubsub.core.PubsubTemplate;
+import org.springframework.cloud.gcp.pubsub.support.GcpHeaders;
+import org.springframework.cloud.gcp.pubsub.support.SubscriberFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.integration.annotation.IntegrationComponentScan;
+import org.springframework.integration.annotation.MessagingGateway;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.integration.channel.PublishSubscribeChannel;
+import org.springframework.integration.gcp.AckMode;
+import org.springframework.integration.gcp.inbound.PubsubInboundChannelAdapter;
+import org.springframework.integration.gcp.outbound.PubsubMessageHandler;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.view.RedirectView;
+
+@SpringBootApplication
+@IntegrationComponentScan
+@RestController
+@ComponentScan(basePackages = {"org.springframework.cloud.gcp"})
```

Comment:
We don't need `@IntegrationComponentScan` since Boot `1.5` any more.
Also it isn't clear why do you need that `@ComponentScan`.
If you don't see some beans from the Spring-GCP project, then some auto-configuration is missed.
Should be fixed there.

Remember `META-INF/spring.factories` with the entry `org.springframework.boot.autoconfigure.EnableAutoConfiguration=\` ?